File: hrms/hr/report/employee_due_date_report/employee_due_date_report.js
// Copyright (c) 2024, Frappe Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt

frappe.query_reports["Employee Due Date Report"] = {
	"filters": [
		{
			"fieldname": "uinput",
			"label": ("Options"),
			"fieldtype": "Select",
			"width": "80",
			"options": [
				 { "value": "promotion_due_date", "label": __("Promotion Due Date") },
				 { "value": "contract_end_date", "label": __("Contact Renew Date") },
				 { "value": "date_of_retirement", "label": __("Retirement Date") },
			],
			"reqd": 1
		},

		{
			"fieldname": "branch",
			"label": ("Branch"),
			"fieldtype": "Link",
			"width": "80",
			"options": "Branch",
		},
		{
			"fieldname":"from_date",
			"label": ("From Date"),
			"fieldtype": "Date",
			"width": "80",

		},
		{
			"fieldname":"to_date",
			"label": ("To Date"),
			"fieldtype": "Date",
			"width": "80",
		},

	]
}

File: hrms/hr/report/employee_due_date_report/employee_due_date_report.py
# Copyright (c) 2024, Frappe Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt

# import frappe
import frappe

def execute(filters=None):
	columns = get_columns(filters)
	data = get_data(filters)
	#filters = get_filters(filters)

	return columns, data

def get_columns(filters):
	cols = [
		("Employee No.") + ":data:150",
		("Employee Name") + ":data:150",
		("Branch")+":data:150",
		("Department")+":data:150",
		("Division")+":data:150",
	]
	if filters.uinput == "promotion_due_date":
		cols.append(("Promotion Due Date")+":date:170")
	if filters.uinput == "contract_end_date":
		cols.append(("Contract End Date")+":date:170")
	if filters.uinput == "date_of_retirement":
		cols.append(("Date of Retirement")+":date:170")
	return cols

def get_data(filters):
	data = "select name, employee_name, branch, department,division, " + str(filters.uinput) +" FROM `tabEmployee` where status = 'Active' and " + str(filters.uinput) + " is not null"
	if filters.get("branch"):
		data += " and branch = \'" + str(filters.branch) + "\'"
	if filters.get("from_date") and filters.get("to_date"):
		data += " and " + str(filters.uinput) +" between \'" + str(filters.from_date) + "\' and \'"+ str(filters.to_date) + "\'"
	return frappe.db.sql(data)
